packages/hop-node/src/watchers/ConfirmRootsWatcher.ts

```diff
@@ -233,16 +234,27 @@ class ConfirmRootsWatcher extends BaseWatcher {
     })
 
     logger.debug(`handling confirmable transfer root ${transferRootHash}, destination ${destinationChainId}, amount ${totalAmount.toString()}, committedAt ${committedAt}`)
-    await this.confirmRootsViaWrapper({
-      rootHashes: [transferRootHash],
-      destinationChainIds: [destinationChainId],
-      totalAmounts: [totalAmount],
-      rootCommittedAts: [committedAt]
-    })
+    try {
+      await this.confirmRootsViaWrapper({
+        rootHashes: [transferRootHash],
+        destinationChainIds: [destinationChainId],
+        totalAmounts: [totalAmount],
+        rootCommittedAts: [committedAt]
+      })
+    } catch (err) {
+      logger.error('confirmRootsViaWrapper error:', err.message)
+      if (err instanceof PreTransactionValidationError) {
+        logger.error('pre transaction validation error. turning off writes.')
+        globalConfig.emergencyDryMode = true
```

Comment:
suggestion: use setter function for setting global config values

```suggestion
        enableEmergencyDryMode()
```

in config.ts

```ts
export function enableEmergencyMode() {
 config.emergencyDryMode = true
}
```